src/tidecli/utils/file_handler.py

```diff
@@ -287,10 +289,13 @@ def get_task_file_data(file_path: Path, metadata: TaskData) -> list[TaskFile]:
                     else:
                         logger.debug("Gap-type exercise answer not valid.")
 
-                        # Validator complains about answer.
-                        # Answer is submitted despite of complains.
+                    if validate_answer_file(answer_bycode, metadata_bycode):
+                        # TODO: tarvitaan lisää testitapauksia,
+                        logger.info("Gap-type exercise answer file is valid.")
                         f1.content = "\n".join(answer_gapcode)
-                        # return []
+                    else:
+                        logger.debug("Gap-type exercise answer not valid.")
```

Comment:
missing file content update will cause wrong content to be sent

Reply:
Doesn't this reintroduce the bug as it was prior to the fix in main?
Also, the if-else should not be duplicated.

For reference, the fixed code in main is currently:

```
                    if validate_answer_file(answer_bycode, metadata_bycode):
                        # TODO: tarvitaan lisää testitapauksia,
                        # Validator OK
                        logger.info("Gap-type exercise answer file is valid.")
                        f1.content = "\n".join(answer_gapcode)
                    else:
                        logger.debug("Gap-type exercise answer not valid.")

                        # Validator complains about answer.
                        # Answer is submitted despite of complains.
                        f1.content = "\n".join(answer_gapcode)
                        # return []
```